### DISCUSSION TITLE
Using php vips to convert indexed/alpha transparent gif/png to jpg with non black background

### DISCUSSION THREAD

**Discussion by richardsyeo** [libvips/php-vips#157] [Q&A]
I have tried without success to use php vips to convert indexed/alpha transparent gif/png to jpg and set background to be color other than black which is default when using below code.

$image = Vips\Image::newFromFile('vips_test_source/quad_indexed.png');
# how do i remove transparency and set background to be white not black?
$image->writeToFile('vips_test_output/quad_indexed.jpg');

Using Ubuntu 22.04 which appears to come with 8.12.1 of vips.

Any solutions much appreciated.

**Comment by jcupitt** [comment]
Hi @richardsyeo, you just set `background`, eg.:

```php
$image->writeToFile('vips_test_output/quad_indexed.jpg', [
    'background' => 255
]); 
```

**Reply by richardsyeo** [reply]
thanks for super quick response. wow thats easy. i was looking for too complicated a solution.
